### PR TITLE
test(SemVer): refine tests and remove unnecessary tests which test th…

### PR DESCRIPTION
**Pull request by bearomorphism**
…e behavior of library

## Checklist

- [ ] I have read the [contributing guidelines](https://commitizen-tools.github.io/commitizen/contributing/)

### Code Changes

- [ ] Add test cases to all the changes you introduce
- [ ] Run `poetry all` locally to ensure this change passes linter check and tests
- [ ] Manually test the changes:
  - [ ] Verify the feature/bug fix works as expected in real-world scenarios
  - [ ] Test edge cases and error conditions
  - [ ] Ensure backward compatibility is maintained
  - [ ] Document any manual testing steps performed
- [ ] Update the documentation for the changes

### Documentation Changes

- [ ] Run `poetry doc` locally to ensure the documentation pages renders correctly
- [ ] Check and fix any broken links (internal or external) in the documentation

> When running `poetry doc`, any broken internal documentation links will be reported in the console output like this:
>
> ```text
> INFO    -  Doc file 'config.md' contains a link 'commands/bump.md#-post_bump_hooks', but the doc 'commands/bump.md' does not contain an anchor '#-post_bump_hooks'.
> ```

### CODE REVIEW

File: tests/test_version_scheme_semver2.py [hunk 1]
```diff
@@ -136,76 +121,3 @@ def test_semver_scheme_property():
 
 def test_semver_implement_version_protocol():
     assert isinstance(SemVer2("0.0.1"), VersionProtocol)
-
-
-def test_semver_sortable():
```
Comment (bearomorphism) [inline tests/test_version_scheme_semver2.py:141]: I'm not sure why we need to test this. The `__lt__` operator is defined in `packaging` library and we are not overriding it.

File: tests/utils.py [hunk 2]
```diff
@@ -21,6 +23,15 @@
 )
 
 
+class VersionSchemeTestArgs(NamedTuple):
+    current_version: str
+    increment: Increment | None
+    prerelease: Prerelease | None
+    prerelease_offset: int
+    devrelease: int | None
+    expected_version: str
```
Comment (bearomorphism) [inline tests/utils.py]: Main change

File: tests/test_version_scheme_semver2.py [hunk 3]
```diff
@@ -1,131 +1,539 @@
...
-]
+from tests.utils import VersionSchemeTestArgs
 
 
 @pytest.mark.parametrize(
-    "test_input, expected",
-    itertools.chain(tdd_cases, weird_cases, simple_flow, linear_prerelease_cases),
+    "args",
+    [
+        # test driven development
```
Comment (Lee-W) [inline tests/test_version_scheme_semver2.py]:
```suggestion
```

I think we can remove this line. This does not provide much information

File: tests/test_version_scheme_semver2.py [hunk 4]
```diff
@@ -1,131 +1,539 @@
...
+        VersionSchemeTestArgs(
+            current_version="0.1.1",
+            increment="PATCH",
+            prerelease=None,
+            prerelease_offset=0,
+            devrelease=None,
+            expected_version="0.1.2",
+        ),
```
Comment (Lee-W) [inline tests/test_version_scheme_semver2.py]: I feel like we should make it something like this

"version_args, expected_version"

```suggestion
        (
            VersionSchemeTestArgs(
                current_version="0.1.1",
                increment="PATCH",
                prerelease=None,
                prerelease_offset=0,
                devrelease=None,,
            ),
        "0.1.2"
```

The expected version probably should be viewed as a different argument

Comment (bearomorphism) [reply tests/test_version_scheme_semver2.py]: sounds good

File: tests/test_version_scheme_semver2.py (hunk 3 above)
Comment (bearomorphism) [reply tests/test_version_scheme_semver2.py]: yep

removed
